runtime/parachains/src/scheduler.rs

```diff
@@ -550,7 +553,11 @@ impl<T: Trait> Module<T> {
 
 		let session_start = <SessionStartBlock<T>>::get();
 		let blocks_since_session_start = now.saturating_sub(session_start);
-		let blocks_since_last_rotation = blocks_since_session_start % config.parachain_rotation_frequency;
+		if config.group_rotation_frequency.is_zero() { 
+			// if there are no rotations, availability timeouts can occur at any block
+			return Some(Box::new(move |_: CoreIndex, _| true));
```

Comment:
The desired behavior is for it to return `true` only if the core has been occupied for too long. This is based on the availability periods. The only difference is that we go without the restriction that availability time-outs can occur only shortly after rotations - because with a rotation frequency of `0`, there aren't any rotations!